src/Components/DataViews/DataViewTable.tsx

```diff
@@ -0,0 +1,59 @@
+import { DataViews } from '@wordpress/dataviews/wp';
+import { Slot } from "@wordpress/components";
+import { ViewportDimensions } from '@/Hooks/ViewportDimensions';
+import type { Action, Field, SupportedLayouts, View } from "@wordpress/dataviews/src/types";
+import type { ReactNode } from "react";
+import './style.scss';
+
+type ItemWithId = { id: string };
+
+type DataViewsProps< Item > = {
+    view: View;
+    namespace: string;
+    onChangeView: ( view: View ) => void;
+    fields: Field< Item >[];
+    search?: boolean;
+    searchLabel?: string;
+    actions?: Action< Item >[];
+    data: Item[];
+    isLoading?: boolean;
+    paginationInfo: {
+        totalItems: number;
+        totalPages: number;
+    };
+    ViewportDimensions?: typeof ViewportDimensions;
+    defaultLayouts: SupportedLayouts;
+    selection?: string[];
+    onChangeSelection?: ( items: string[] ) => void;
+    onClickItem?: ( item: Item ) => void;
+    isItemClickable?: ( item: Item ) => boolean;
+    header?: ReactNode;
+} & ( Item extends ItemWithId
+    ? { getItemId?: ( item: Item ) => string }
+    : { getItemId: ( item: Item ) => string } );
+
+const applyFiltersToTableElements = (namespace: string, elementName: string, element) => {
+    return wp.hooks.applyFilters( `dokan_${namespace.toLowerCase()}_dataviews_${elementName}`, element );
+};
+
+const DataViewTable = ( props: DataViewsProps< Item > ) => {
+    const filteredProps = {
+        ...props,
+        data: applyFiltersToTableElements( props?.namespace, 'data', props.data ),
+        view: applyFiltersToTableElements( props?.namespace, 'view', props.view ),
+        fields: applyFiltersToTableElements( props?.namespace, 'fields', props.fields ),
+        actions: applyFiltersToTableElements( props?.namespace, 'actions', props.actions ),
+    };
+
+    return (
+        <div className="dokan-dashboard-datatable">
```

Comment:
We will add an `id` attribute by converting the namespace to **kebabCase** so that anyone can find and  predict the filters e.i `my-table`

> **Ref:** we can use change-case package https://www.npmjs.com/package/change-case and also see the [WooCommerce uses](https://github.com/woocommerce/woocommerce/blob/f7d099f6eeafa1fdae39125466fadc7318b89c13/plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/product-details/index.tsx#L4)